### DISCUSSION TITLE
Don't want HTTPS

### DISCUSSION THREAD

**Discussion by jfmatth** [piku/piku#312] [General]
Love this project, it's super neat, but I can't seem to get it started :(

The ENV docs seem to be incomplete, since the sample python app has settings not defined in the ENV docs, namely 
```
NGINX_SERVER_NAME
NGINX_IPV4_ADDRESS
```

Also, how do I use nip.io as my web address?  I want to use this on a home lab setup, but can't seem to get it going.

Any help is appreciated.

Ill do what I can for PR's against the docs too.

**Comment by jfmatth** [comment]
In looking over the piku.py code, which is very easy to read.  I notice that if the NGINX_SERVER_NAME is configured, it automatically creates a lets encrypt cert, which may be the case sometimes, but I don't want it at home.

I'm I using piku right for home use, or should I not use the server name?

**Reply by rcarmo** [reply]
You can use the server name just fine, although at home it will be more or less useful depending on your particular LAN setup. I have apps that just use ports, and others that use `machine.local` (because Debian and Ubuntu announce those and macOS and iOS can resolve them). Other people have home routers that assign names like `machine.lan` automatically, and you can use those as well and just ignore HTTPS.

**Reply by jfmatth** [reply]
thanks @rcarmo 

That's fine and all, but I don't want HTTPS and it looks like just assigning a server name creates the cert, etc.  Which I don't want.

I've got a pretty standard setup with proxmox VM's running piku and a home router to the internet.  There is no name conversion or anything like that going on.

If my IP of the server is 192.168.1.10, then python.192.168.1.10.nip.io will resolve to that server with the appropriate host header I need for NGINX.  What's not working is the cert, since it fails in the generation.

I tried this on a Digital Ocean droplet and it worked perfect with NIP.IO, so i know it's not that part.

Hope this makes sense.

**Reply by chr15m** [reply]
@jfmatth if the certificate generation fails it will use a self-signed certificate I think and then you can just ignore it and use http to load your site. Make sure you don't have `HTTPS_ONLY` set as that will create a redirect from http to https.

**Reply by rcarmo** [reply]
OK, but the default config will let you use HTTP instead of HTTPs. You just need to [set `NGINX_HTTPS_ONLY` to false](https://github.com/piku/piku/blob/master/piku.py#L924C37-L924C53).

So the certificate will be generated (it will [fall back to a self-signed one](https://github.com/piku/piku/blob/master/piku.py#L790)) but you can access your service via HTTP and ignore it altogether (which is what I do on my LAN with Proxmox as well - hence I don't know why that wouldn't work for you).

**Reply by jfmatth** [reply]
Hi all, and thank you so much for the help, but I'm still not working :(

I tried deploying the sample python app with

NGINX_SERVER_NAME=python.192.168.100.216.nip.io
NGINX_HTTPS_ONLY=false

Then going to the site, i get bad gateway.

The NGINX logs show that uwsgi isn't responding :(

```
jfmatth@piku:~$ cat /var/log/nginx/error.log
2023/07/07 13:33:24 [notice] 21833#21833: signal process started
2023/07/07 13:33:26 [notice] 21837#21837: signal process started
2023/07/07 13:34:08 [emerg] 23178#23178: cannot load certificate "/home/piku/.piku/nginx/sample_python_app.crt": PEM_read_bio_X509_AUX() failed (SSL: error:0480006C:PEM routines::no start line:Expecting: TRUSTED CERTIFICATE)
2023/07/07 13:34:10 [notice] 23180#23180: signal process started
2023/07/07 13:34:52 [emerg] 24527#24527: cannot load certificate "/home/piku/.piku/nginx/sample_python_app.crt": PEM_read_bio_X509_AUX() failed (SSL: error:0480006C:PEM routines::no start line:Expecting: TRUSTED CERTIFICATE)
2023/07/07 13:34:53 [notice] 24531#24531: signal process started
2023/07/07 13:35:49 [crit] 24535#24535: *9 connect() to unix:///home/piku/.piku/nginx/sample_python_app.sock failed (2: No such file or directory) while connecting to upstream, client: 192.168.100.152, server: python.192.168.100.216.nip.io, request: "GET / HTTP/1.1", upstream: "uwsgi://unix:///home/piku/.piku/nginx/sample_python_app.sock:", host: "python.192.168.100.216.nip.io"
2023/07/07 13:35:49 [crit] 24535#24535: *9 connect() to unix:///home/piku/.piku/nginx/sample_python_app.sock failed (2: No such file or directory) while connecting to upstream, client: 192.168.100.152, server: python.192.168.100.216.nip.io, request: "GET /favicon.ico HTTP/1.1", upstream: "uwsgi://unix:///home/piku/.piku/nginx/sample_python_app.sock:", host: "python.192.168.100.216.nip.io", referrer: "http://python.192.168.100.216.nip.io/"
```

there is no file in home/piku/.piku/nginx/sample_python_app.sock

Anything else I should check?

**Comment by jfmatth** [comment]
Hello everyone, I wanted to update this discussion and let you know I found my issue.

It boils down to the address I was using for the site.  Since I'm internal and don't have DNS setup for my environment, I use nip.io, which maps to IP's based on the DNS name.  I was using the <IP>.nip.io as the website and it kept breaking.  I have to prefix that with anything to make it work, i.e. web.<IP>.nip.io and it works fine.  HTTP and HTTPS work as advertised.

I'm very excited to get PIKU working since it's just what I've wanted for a while.

Thanks everyone for helping me with this.
